python_modules/dagster/dagster/_core/definitions/time_window_partition_mapping.py

```diff
@@ -397,6 +397,12 @@ def _offsetted_datetime(
 ) -> datetime:
     if partitions_def.is_basic_daily and offset != 0:
         result = dt + timedelta(days=offset)
+
+        # Due to daylight savings time, adding a day might not result in the same hour
+        # Override the hour to match the original datetime
+        if result.hour != dt.hour:
+            result = result.replace(hour=dt.hour)
```

Comment:
can we fall back to the 'slow' logic below instead of trying to correct it ourselves (since i think this could leave us a day behind if we got moved from e.g. midnight to 11PM on the previous day)